scripts/tfw_lib.sh

```diff
@@ -36,27 +36,32 @@ calc()
 distribute_queues()
 {
 	dev=$1
-	RXQ_MAX=$(ethtool -l $dev 2>/dev/null \
-		  | grep -m 1 RX | sed -e 's/RX\:\s*//')
+	RXQ_MAX=$2
 
-	echo "...distribute $dev queues"
+	echo "...set rx channels to $RXQ_MAX, please wait..."
+	# Set maximum number of available channels for better
+	# packets hashing.
+	ethtool -L $dev rx $RXQ_MAX >/dev/null 2>&1
+	# Wait for the interface reconfiguration.
+	opstate="$TFW_NETDEV_PATH/$dev/operstate"
+	while [ "$(cat $opstate)" = "down" ]; do
+		sleep 1
+	done
+
+	dev_irqs_path="/sys/class/net/$dev/device/msi_irqs"
+	irqs=($(grep $dev /proc/interrupts | sed -e 's/\s*\|:.*//g'))
+	if [ -z "$irqs" -a -d $dev_irqs_path ]; then
+		irqs=($(ls $dev_irqs_path))
```

Comment:
The things are going complex, so please write a short (as above) comment why do we need the 2 sources to learn interrupts.

Reply:
Comment added.